runtime/j9jitbuilder/JavaMethodCallbackGen.cpp

```diff
@@ -19,23 +19,272 @@
+	if (_returnType == _vm->voidReflectClass) {
+		functionAddress = (void*) _vmThread->functions->CallVoidMethodA;
+	} else if (_returnType == _vm->booleanReflectClass) {
+		functionAddress = (void*) _vmThread->functions->CallBooleanMethodA;
+	} else if (_returnType == _vm->charReflectClass) {
+		functionAddress = (void*) _vmThread->functions->CallCharMethodA;
+	} else if (_returnType == _vm->floatReflectClass) {
+		functionAddress = (void*) _vmThread->functions->CallFloatMethodA;
+	} else if (_returnType == _vm->doubleReflectClass) {
+		functionAddress = (void*) _vmThread->functions->CallDoubleMethodA;
+	} else if (_returnType == _vm->byteReflectClass) {
+		functionAddress = (void*) _vmThread->functions->CallByteMethodA;
+	} else if (_returnType == _vm->shortReflectClass) {
+		functionAddress = (void*) _vmThread->functions->CallShortMethodA;
+	} else if (_returnType == _vm->intReflectClass) {
+		functionAddress = (void*) _vmThread->functions->CallIntMethodA;
+	} else if (_returnType == _vm->longReflectClass) {
+		functionAddress = (void*) _vmThread->functions->CallLongMethodA;
+	} else {
+		Assert_J9JITBUILDER_unreachable();
+	}
+
+	return functionAddress;
+}
+
+char*
+JavaMethodCallbackGen::getCallInFunctionName()
+{
+	char* name = NULL;
+	if (_returnType == _vm->voidReflectClass) {
+		name = (char*) "CallVoidMethodA";
+	} else if (_returnType == _vm->booleanReflectClass) {
+		name = (char*) "CallBooleanMethodA";
+	} else if (_returnType == _vm->charReflectClass) {
+		name = (char*) "CallCharMethodA";
+	} else if (_returnType == _vm->floatReflectClass) {
+		name = (char*) "CallFloatMethodA";
+	} else if (_returnType == _vm->doubleReflectClass) {
+		name = (char*) "CallDoubleMethodA";
+	} else if (_returnType == _vm->byteReflectClass) {
+		name = (char*) "CallByteMethodA";
+	} else if (_returnType == _vm->shortReflectClass) {
+		name = (char*) "CallShortMethodA";
+	} else if (_returnType == _vm->intReflectClass) {
+		name = (char*) "CallIntMethodA";
+	} else if (_returnType == _vm->longReflectClass) {
+		name = (char*) "CallLongMethodA";
+	} else {
+		Assert_J9JITBUILDER_unreachable();
+	}
+	return name;
+}
+
+char*
+JavaMethodCallbackGen::getSignatureFromJ9Class(J9Class *type)
+{
+	char* name = NULL;
+	if (type == _vm->voidReflectClass) {
+		name = (char*) "V";
+	} else if (type == _vm->booleanReflectClass) {
+		name = (char*) "Z";
+	} else if (type == _vm->charReflectClass) {
+		name = (char*) "C";
+	} else if (type == _vm->floatReflectClass) {
+		name = (char*) "F";
+	} else if (type == _vm->doubleReflectClass) {
+		name = (char*) "D";
+	} else if (type == _vm->byteReflectClass) {
+		name = (char*) "B";
+	} else if (type == _vm->shortReflectClass) {
+		name = (char*) "S";
+	} else if (type == _vm->intReflectClass) {
+		name = (char*) "I";
+	} else if (type == _vm->longReflectClass) {
+		name = (char*) "J";
+	} else {
+		Assert_J9JITBUILDER_unreachable();
+	}
+	return name;
+}
+
+char*
+JavaMethodCallbackGen::getCallInFunctionLineNumber()
```

Comment:
Needs a lot of doc to describe why these numbers, and why they're necessary at all